Jellyfin.Api/Controllers/VideosController.cs

```diff
@@ -425,7 +425,7 @@ [FromQuery] [RegularExpression(EncodingHelper.ValidationRegex)] string? subtitle
             EnableAudioVbrEncoding = enableAudioVbrEncoding
         };
 
-        var state = await StreamingHelpers.GetStreamingState(
+        using var state = await StreamingHelpers.GetStreamingState(
```

Comment:
this will cause the stream to be closed if disposed here resulting in only the first chunk beeing transmitted

Reply:
I was concerned about that, but similar code in [`AudioHelper.GetAudioStream`](https://github.com/jellyfin/jellyfin/blob/81aca67745d541e276cfa05efe57e52e9f5a5d0e/Jellyfin.Api/Helpers/AudioHelper.cs#L88) uses `using`, which appears to be the audio counterpart to `GetVideoStream`. Is that a bug, or is it only an issue for the video case?

Reply:
Videos have transcode throttler which will have the need to manage the state and read from this state during the whole transcoding and this should only be disposed after the transcoding process has being killed or exited.

- StartFfMpeg assign the current TranscodingJob to StreamState
- Once StreamState is disposed, StreamState.TranscodingJob = null
- StreamState.ReportTranscodingProgress() cannot update TranscodingPositionTicks in current TranscodingJob
- job.TranscodingPositionTicks stuck at the initial value
- TranscodingThrottler query job.TranscodingPositionTicks -> IsThrottleAllowed == false

Basically I'm very against modifying code without understanding the consequence and just make "less warnings". Yes warnings sucks but cleanups like this should be made with certain level of confidence of the codebase.

Reply:
I understand that one shouldn't blindly modify stuff for the sole purpose of less warning noise, but I thought looking at some of the more straightforward warnings would be a small start to getting familiarity with the codebase for future work. I chose warnings regarding `IDisposable` as I read that they could perhaps be related to potential memory leaks in some scenarios.

I did leave half of the CA2000 warnings alone in part due to complexity; I was going to leave this one as well, but I saw **extremely** similar code in `GetAudioStream` and naively thought it would apply here as well. I'm sorry about that.

What is the simplest way to test the `GetVideoStream` path for transcodes? I've been playing around with transcode/network settings and transcoding appears to use the `DynamicHlsController` path.